Code/Framework/AzCore/AzCore/Asset/AssetManager.cpp

```diff
@@ -984,7 +985,7 @@ namespace AZ::Data
             }
             else
             {
-                AZ_Warning("AssetManager", false, "GetAsset called for asset which does not exist in asset catalog and cannot be loaded.  Asset may be missing, not processed or moved.  AssetId: %s",
+                AZLOG_WARN("GetAsset called for asset which does not exist in asset catalogand cannot be loaded.Asset may be missing, not processed or moved.AssetId: % s",
```

Comment:
@lawsonamzn looking for your opinion on this one

Reply:
Why the removal of spaces and compaction?

Note:  This is probably the wrong place to make changes to reduce noise.  This AssetManager serves the entire engine, the graphics, audio, animation, levels, everything, and this function is used to warn that an asset is asked for that doesn't exist (or doesn't exist yet).  Removing it from view is going to be unhelpful when the real solution is generally to either NOT call GetAsset unless you're sure such an asset exists, or, to use the Asset PRocessor Api (like CompileAssetSync) to ensure that the Asset Processor generates the asset if its in the queue, before you call GetAsset.

Reply:
In this particular situation, a system receives a SourceFileChanged notification for a file it knows will get processed by the AP. So then it connects to the asset bus on the source file uuid, and calls queue load to receive the OnAssetReady (or Error) calls back. If it can recieve those notifications without trying to load the asset, I will do that, but that's how I thought this had to happen.

Reply:
removal of whitespace and compaction was just a copy paste error.

Reply:
When SourceFileChanged happens, it means the source file has changed but not necessarily that a product is ready.  It can take time to process it.  A different message is available if you want to listen for assets changing in the cache.

In fact, trying to load an _asset_ when a _source_ changes is a recipe for problems because you'll almost always get Source File Changed some time before the asset exists (especially on first compile).  

Note that this error/warning message happens when it cannot even find the asset in the catalog - ie, it does not exist (yet).   Its usually an error to request something that does not exist.

In the engine, Asset<T> is generally mean to reference an in memory loaded asset.  If you want to work with the asset system API on the file level (instead of on the asset level), then you can hook into the catalog bus broadcasts, such as AssetCatalogEvents OnCatalogAssetChanged/OnCatalogAssetAdded, which does not require you to subscribe to an address and will tell you about every asset change.  These happen at the file and catalog level, wheras OnAssetready / OnAssetReloaded is specifically for assets that are already loaded or being loaded.

Note that AssetIds are just SourceFileID + SubId typles, so if you know a source id, you can listen for its assets to be done by keeping track of what sources you're looking out for and listening ton OnCatalogAssetChanged/OnCatalogAssetAdded.  

So basically, if you just want to know about changes in asset data on the file system, use the AssetCatalogEvents bus.  If you have _already_ loaded an asset (so you already have an Asset<T>) you can use the OnAssetReady / OnAssetReloaded to track changes to that specific instance of that specific asset.    The full sequence of events for a not-currently-loaded asset is something like

OnSourceFileChanged
...some time passes ..
 OnCatalogAssetChanged/OnCatalogAssetAdded (when the products become available and have been compiled)
  ... some more time passes..
OnAssetReloaded/OnAssetReady (if and only if there is a currently loaded instance of that asset<T> in memory and also only if the system that owns that asset type has opted into default behavior for reloading).

Reply:
Cool, I'll work with the catalog events.